contracts/chain-adapters/Scroll_Adapter.sol

```diff
@@ -0,0 +1,77 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import "@scroll-tech/contracts/L1/gateways/IL1GatewayRouter.sol";
+import "@scroll-tech/contracts/L1/IL1ScrollMessenger.sol";
+import "./interfaces/AdapterInterface.sol";
+
+contract Scroll_Adapter is AdapterInterface {
+    using SafeERC20 for IERC20;
+    uint32 public immutable l2GasLimit = 250_000;
+
+    IL1GatewayRouter public immutable l1GatewayRouter;
+    IL1ScrollMessenger public immutable l1ScrollMessenger;
+
+    /**
+     * @notice Constructs new Adapter.
+     * @param _l1GatewayRouter Standard bridge contract.
+     * @param _l1ScrollMessenger Scroll Messenger contract.
+     */
+    constructor(IL1GatewayRouter _l1GatewayRouter, IL1ScrollMessenger _l1ScrollMessenger) {
+        l1GatewayRouter = _l1GatewayRouter;
+        l1ScrollMessenger = _l1ScrollMessenger;
+    }
+
+    /**
+     * @notice Send message to `target` on L2.
+     * @dev This method is marked payable because relaying the message might require a fee
+     * to be paid by the sender to forward the message to L2. However, it will not send msg.value
+     * to the target contract on L2.
+     * @param target L2 address to send message to.
+     * @param message Message to send to `target`.
+     */
+    function relayMessage(address target, bytes calldata message) external payable {
+        // We can specifically send a message with 0 value to the Scroll Bridge
+        // and it will not forward any ETH to the target contract on L2. However,
+        // we need to set the payable value to msg.value to ensure that the Scroll
+        // Bridge has enough gas to forward the message to L2.
+        l1ScrollMessenger.sendMessage{ value: msg.value }(target, 0, message, l2GasLimit);
```

Comment:
What happens with the excess value if we set the second param to 0? Do we get refunded on L1?

Reply:
Correct - the bridge calls our fallback function with the excess value.

Reply:
where is that documented OOC?

Reply:
It's documented here: https://docs.scroll.io/en/developers/l1-and-l2-bridging/eth-and-erc20-token-bridge/

```
depositETH and depositERC20 are payable functions, the amount of ETH sent to these functions will be used to pay for L2 fees. If the amount is not enough, the transaction will not be sent. All excess ETH will be sent back to the sender. 0.00001 ETH should be more than enough to process a token deposit.
```

Reply:
can you document this link in the code if not already?

Reply:
Since adapters and chain specific spoke pools are network specific I like leaving links to technical documentation to help with future upgrades

Reply:
There are comments in the code that give an explainer, but I added an actual link here: https://github.com/across-protocol/contracts-v2/pull/381/commits/272245aeacfc9c7d9387444428a5fbef2e972628